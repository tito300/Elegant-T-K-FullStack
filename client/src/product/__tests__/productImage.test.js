import { shallow } from 'enzyme';
import React from 'react';
import ProductImage from '../productImage';

let wrapper;

describe('ProductImage Component', ()=> {
    beforeAll(()=>{
        let props = {
            product: {secondaryPhotos: ['img/1.jpg', 'img/2.jpg']}, 
            activePicture: 'randompicture', 
            handleSelectPhoto: jest.fn(), 
        }
        wrapper = shallow(<ProductImage {...props} />);
    })
    it('should match snapshot', ()=>{
        expect(wrapper).toMatchSnapshot();
    })
    it('should display default photo if no secondary is available', ()=>{
        wrapper.setProps({
            product: {secondaryPhotos: []},
            activePicture: 'defaultphoto',
        })
        expect(wrapper.find('.galleryPhoto').prop('src')).toContain('defaultphoto')
    })
})