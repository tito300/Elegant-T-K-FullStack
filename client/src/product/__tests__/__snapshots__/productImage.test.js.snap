// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Component should match snapshot 1`] = `
<div
  className="photoSection"
>
  <div
    className="photoContainer"
  >
    <img
      alt=""
      className="photo"
      src="/randompicture"
    />
  </div>
  <div
    className="gallerySection"
  >
    <div>
      <i
        aria-hidden="true"
        className="fa fa-angle-left"
      />
    </div>
    <div
      className="galleryContainer"
    >
      <div
        className="imgContainer"
        key="0"
      >
        <img
          alt="product"
          className="galleryPhoto"
          onClick={[MockFunction]}
          photoindex={0}
          src="/img/1.jpg"
        />
      </div>
      <div
        className="imgContainer"
        key="1"
      >
        <img
          alt="product"
          className="galleryPhoto"
          onClick={[MockFunction]}
          photoindex={1}
          src="/img/2.jpg"
        />
      </div>
    </div>
    <div>
      <i
        aria-hidden="true"
        className="fa fa-angle-right"
      />
    </div>
  </div>
</div>
`;
